### ISSUE TITLE
Update ansible playbooks to use up-to-date version of git 

### ISSUE THREAD

**Issue by sej-jackson** [adoptium/infrastructure#2889]
The Git project has now released new versions of git to fix recent security vulnerabilities, and since the playbooks are installing git built from source of version 2.15.0 (from October 2017), changing them to use the fixed version (2.39.1) would help with ensuring that the systems we configure with them are fixed also.  

https://github.blog/2023-01-17-git-security-vulnerabilities-announced-2/

Searching the playbooks for "git" hits on 97 files most of which are likely to be unrelated to the installation of git itself, but will still need checking. However, searching on "git --version" found these 5, all of which will definitely need updating:
```
% find . -type f -exec grep -li "git --version" {} \;
./AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
./AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/SLES.yml
./AdoptOpenJDK_Unix_Playbook/roles/GIT_Source/tasks/main.yml
./vagrant.yml
./AdoptOpenJDK_ITW_Playbook/roles/GIT_Source/tasks/main.yml
% 
```

I checked the Dockerfiles also, but they don't include any hard-coded git versions.

**Comment by aswinkr77** [comment]
Compiled and installed `git 2.41.0` successfully on the docker centos6 image pulled from [adoptopenjdk/centos6_build_image](https://hub.docker.com/r/adoptopenjdk/centos6_build_image). Needs the `libcurl-devel` package for compiling git from source.

**Comment by aswinkr77** [comment]
Looks like compiling git version >2.26 requires the `libcurl-devel` package on all distros(name of the package maybe different on different distros).

**Comment by sxa** [comment]
Yeah on the .deb platforms (Debian, Ubuntu etc.) it's a slightly less consistent package name that has a version number in it so it will potentially require further investigation to get it right everywhere.

**Comment by aswinkr77** [comment]
@sxa The dependency package is already available in the `Common` role for CentOS, Rhel, Ubuntu and SLES.

Actually, I tested the latest git version on the Adopitum centos6 build image pulled from the docker image which hasn't been updated for over 9 months(related issue #3211) and was missing the `libcurl-devel` package but in reality, the package is available in the Unix Playbook.

Note: I tested the latest version `2.41` in CentOS, Rhel, Ubuntu, SLES 15 and SLES 12.5 with the dependency package installed.

For CentOS:
`libcurl-devel `
https://github.com/adoptium/infrastructure/blob/fcb6021ff1afbd3ff208586217929ae2f38f8225/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/CentOS.yml#L100-L107

For Rhel:
`libcurl-devel`
https://github.com/adoptium/infrastructure/blob/fcb6021ff1afbd3ff208586217929ae2f38f8225/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/RedHat.yml#L110-L114

For Ubuntu:
`libcurl4-openssl-dev`
https://github.com/adoptium/infrastructure/blob/fcb6021ff1afbd3ff208586217929ae2f38f8225/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Ubuntu.yml#L23-L26

For SLES 15:
`libcurl-devel`
https://github.com/adoptium/infrastructure/blob/fcb6021ff1afbd3ff208586217929ae2f38f8225/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/SLES.yml#L31-L36

For SLES 12.5:
`libcurl-devel`
https://github.com/adoptium/infrastructure/blob/fcb6021ff1afbd3ff208586217929ae2f38f8225/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/SLES.yml#L70-L71

For Fedora:
`libcurl-devel`
https://github.com/adoptium/infrastructure/blob/fcb6021ff1afbd3ff208586217929ae2f38f8225/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Fedora.yml#L29-L33

For openSUSE:
`libcurl-devel`
https://github.com/adoptium/infrastructure/blob/fcb6021ff1afbd3ff208586217929ae2f38f8225/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/openSUSE.yml#L19-L23

For Debian:
`libcurl4-openssl-dev`
https://github.com/adoptium/infrastructure/blob/fcb6021ff1afbd3ff208586217929ae2f38f8225/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Debian.yml#L20-L24
